cluster/src/main/java/io/scalecube/cluster/ClusterMembership.java

```diff
@@ -68,11 +71,14 @@
   private static final Func1<Message, Boolean> GOSSIP_MEMBERSHIP_FILTER = new Func1<Message, Boolean>() {
     @Override
     public Boolean call(Message message) {
-      Object data = message.data();
-      return data != null && ClusterMembershipData.class.equals(data.getClass());
+      return message.data() != null && ClusterMembershipData.class.isAssignableFrom(message.data().getClass());
```

Comment:
Why "isAssignableFrom"? We don't need to support superclasses or super interfaces here.